jinahub/indexers/searcher/FaissSearcher/tests/integration/test_train_index.py

```diff
@@ -32,48 +31,35 @@ def metas(tmpdir):
 
 
 def test_train_and_index(metas, tmpdir):
-    vec_idx = np.random.randint(0, high=512, size=[512]).astype(str)
-    vec = np.array(np.random.random([512, 10]), dtype=np.float32)
-
     query = np.array(np.random.random([10, 10]), dtype=np.float32)
     query_docs = _get_docs_from_vecs(query)
 
-    train_data_file = os.path.join(os.environ['TEST_WORKSPACE'], 'train.npy')
-    train_data = np.array(np.random.random([1024, 10]), dtype=np.float32)
-    np.save(train_data_file, train_data)
-
-    trained_index_file = os.path.join(os.environ['TEST_WORKSPACE'], 'faiss.index')
+    trained_index_file = os.path.join(tmpdir, 'faiss.index')
+    train_data = np.array(np.random.random([512, 10]), dtype=np.float32)
 
-    export_dump_streaming(
-        os.path.join(tmpdir, 'dump'),
-        1,
-        len(vec_idx),
-        zip(vec_idx, vec, [b'' for _ in range(len(vec))]),
-    )
-    dump_path = os.path.join(tmpdir, 'dump')
+    index_docs = _get_docs_from_vecs(train_data)
 
     f = Flow().add(
         uses=FaissSearcher,
         timeout_ready=-1,
         uses_with={
-            'index_key': 'IVF10_HNSW32,PQ2',
+            'index_key': 'IVF6,PQ2',
             'trained_index_file': trained_index_file,
         },
+        uses_meta=metas,
     )
-    with f:
-        # the trained index will be dumped to "faiss.index"
-        f.post(on='/train', parameters={'train_data_file': train_data_file})
 
-    f = Flow().add(
-        uses=FaissSearcher,
-        timeout_ready=-1,
-        uses_with={
-            'index_key': 'IVF10_HNSW32,PQ2',
-            'trained_index_file': trained_index_file,
-            'dump_path': dump_path,
-        },
-    )
     with f:
+        import faiss
+
+        faiss_index = faiss.index_factory(10, 'IVF6,PQ2', faiss.METRIC_INNER_PRODUCT)
+        faiss.normalize_L2(train_data)
+        faiss_index.train(train_data)
+        faiss.write_index(faiss_index, trained_index_file)
+
+        # train and index docs first
```

Comment:
Didn't we say that we want the trained index to contain the data as well?

so just

- train outside 
- start Flow -- Faiss gets pretrained model with data
- search on the data is ready